docs/operators/fsInputOperator.md

```diff
@@ -0,0 +1,159 @@
+
+The operator is fault tolerant. It tracks previously read files and current offset as part of checkpoint state. In case of failure the operator will skip files that were already processed and fast forward to the offset of the current file. Supports both static and dynamic partitioning. For static partitioning it also accpets changes to number of partitions at run time. The directory scanner is responsible to only accept the files that belong to a partition.
+## Partitioning of File Input Operator
+#### Static Partitioning of File Input Operator
+Configure parameter "partitionCount" to define desired number of static partitions.
+
+```xml
+  <property>
+    <name>dt.operator.{OperatorName}.prop.partitionCount</name>
+    <value>4</value>
+  </property>
+```
+
+where {OperatorName} is the name of the input operator.
+Above lines will partition operator statically 4 times. Above value can be changed accordingly to change the number of static partitions.
+
+#### Dynamic Partitioning of File Input Operator
+Override ```getNewPartitionCount()``` method to do dymanic partitioning.
+
+File Input Operator is **idempotent**, **fault-tolerant** and **partitionable**.
+
+## Operator Usecase
+1. Read all files of a directory and then keep scanning it for newly added files.
+
+## Operator Information
+1. Operator location: ***malhar-library***
+2. Available since: ***1.0.2***
+3. Operator state: ***Stable***
+3. Java Packages:
+    * Operator: ***[com.datatorrent.lib.io.fs.AbstractFileInputOperator](https://www.datatorrent.com/docs/apidocs/com/datatorrent/lib/io/fs/AbstractFileInputOperator.html)***
+
+### AbstractFileInputOperator
+This is the abstract implementation that serves as base class for scanning a directory for files and read the files one by one. The default implementation scans a single directory, you can extend this class to override scanner to work with multiple input directories. [This](#https://github.com/DataTorrent/examples/tree/master/tutorials/fileIO-multiDir) example in datatorrent examples repository demostrates how to do that. This class doesn’t have any ports.
+
+![AbstractFileInputOperator.png](images/fsInput/operatorsClassDiagram.png)
+
+## Properties and Ports
+### <a name="AbstractFileInputOperatorProps"></a>Properties of AbstractFileInputOperator
+| **Property** | **Description** | **Type** | **Mandatory** | **Default Value** |
+| -------- | ----------- | ---- | ------------------ | ------------- |
+| *directory* | absolute path of directory to be scanned | String | Yes | N/A |
+| *scanIntervalMillis* | Interval in milliseconds after which directory should be scanned for new files | int | No | 5000 |
+| *emitBatchSize* | Number of tuples to emit in a batch. Batch is collection of tuples bundled together. Multiple of such batches will be emitted in a window. | int | No | 1000 |
```

Comment:
The description of a batch is incorrect -- I see no evidence of any "bundling" of a batch and also the explanation
simply explains one undefined term ("batch") using another undefined term ("bundle"). It is also inadequate: Why is batching done here ? Why should the user care about it ? How should the user decide when to change this parameter ?